src/components/course/data/hooks.jsx

```diff
@@ -656,52 +656,57 @@ export const useUserSubsidyApplicableToCourse = ({
     }
 
     const {
-      catalog: {
-        containsContentItems,
-        catalogList: catalogsWithCourse,
-      },
+      catalog: { containsContentItems, catalogList: catalogsWithCourse },
       courseDetails,
     } = courseData;
 
     let applicableUserSubsidy;
 
-    // if course can be redeemed with a subsidy access policy, return `learnerCredit` subsidy type.
-    if (isPolicyRedemptionEnabled) {
-      // the enterprise-access `can-redeem` API returns `can_redeem: false` when a learner
-      // has already redeemed a course. This means the course page thinks the learner no
-      // longer has any subsidy available to spend. `isPolicyRedemptionEnabled` is true when
-      // `can_redeem: false && has_successful_redemption: true`, so `redeemableSubsidyAccessPolicy` may now be null.
-      applicableUserSubsidy = {
-        discountType: 'percentage',
-        discountValue: 100,
-        subsidyType: LEARNER_CREDIT_SUBSIDY_TYPE,
-        perLearnerEnrollmentLimit: redeemableSubsidyAccessPolicy?.perLearnerEnrollmentLimit,
-        perLearnerSpendLimit: redeemableSubsidyAccessPolicy?.perLearnerSpendLimit,
-        policyRedemptionUrl: redeemableSubsidyAccessPolicy?.policyRedemptionUrl,
-      };
-    }
+    const getSubscriptionLicenseSubsidy = async () => {
+      if (!subscriptionLicense) {
+        return null;
+      }
+      try {
+        // get subscription license with extra information (i.e. discount type, discount value, subsidy checksum)
+        const fetchLicenseSubsidyResponse = await courseService.fetchUserLicenseSubsidy();
+        if (fetchLicenseSubsidyResponse) {
+          return camelCaseObject(fetchLicenseSubsidyResponse.data);
+        }
+      } catch (error) {
+        logError(error);
+        if (onSubscriptionLicenseForCourseValidationError) {
+          onSubscriptionLicenseForCourseValidationError(error);
+        }
+      }
+      return null;
+    };
+
+    const setLearnerCreditSubsidy = () => {
```

Comment:
Lets name this consistently with the new function on 665 `getSubscriptionLicenseSubsidy`.